### DISCUSSION TITLE
fatal: the remote end hung up unexpectedly

### DISCUSSION THREAD

**Discussion by mong00se007** [Guilouz/Klipper-Flsun-Speeder-Pad#179] [Q&A]
Hi, 
Hope someone can help. 
I followed the instructions fine. But when it came to install the official builds. It clones klipper up to 21% then it hangs.
Been trying for a few days.  I have tried just cloning it straight, same problem.
Any help is much appreciated! 

p.s. some one else has a similar problem about a month ago, but no reply:
https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/173

----------------------------

###### Cloning Klipper from https://github.com/Klipper3d/klipper ...
Cloning into '/home/pi/klipper'...
remote: Enumerating objects: 63562, done.
remote: Counting objects: 100% (116/116), done.
remote: Compressing objects: 100% (74/74), done.
fatal: the remote end hung up unexpectedly5 MiB | 147.00 KiB/s
fatal: early EOF
fatal: index-pack failed

#=======================================================#
 Cloning Klipper from
 https://github.com/Klipper3d/klipper
 failed!
#=======================================================#
------------------------------------

**Comment by lin-ycv** [comment]
most likely one of the following reasons:
1. github is experiencing server problems
2. your internet is unstable/too slow, and it's timing out.

if you're sure github works fine, try moving your printer closer to your wifi ap

**Reply by mong00se007** [reply]
I'm pretty sure it's something else. All updates to ubuntu work fine. I have a good wifi network and fiber. I've re-installed multiple times and it always hangs at 21% the pad does not disconnect from wifi. Anyway I'll keep trying different things. Unless someone knows what this is.. thank you for your help

**Reply by lin-ycv** [reply]
I stopped getting this error by moving the printer closer to my access point, maybe it's just a coincidence. 
You could try to set the git compression to a different level or modify the script to use depths of one when cloning.

**Reply by mong00se007** [reply]
Got it working, 4th time lucky I guess. I locked the screen to the AP just in case, but this time it failed at 35% I went back into KIAUH and deleted klipper a second time, and then tried again, and it worked. Thanks again for your help!
